src/zenml/client.py

```diff
@@ -1987,6 +1988,7 @@ def list_stack_components(
         type: Optional[str] = None,
         workspace_id: Optional[Union[str, UUID]] = None,
         user_id: Optional[Union[str, UUID]] = None,
+        metadata: Optional[List[str]] = None,
```

Comment:
This should probably be `Dict[str, str]` as in the other places?

Reply:
You're right. For now, I have removed it altogether from the filtering logic.